### PR TITLE
Adding MavenCoordinatesMacroExpander

### PR DESCRIPTION
**Pull request by bocon13**
This expander transforms build rule into maven coordinates associated
with that rule. The build rule must have maven coordinates.

Example:
$(maven_coords :target) --> groupId:artifactId:version

### CODE REVIEW

File: docs/__genrule_common.soy [hunk 1]
```diff
@@ -60,6 +60,12 @@
       <dd>Expands to the location of the output of the build rule. This
       means that you can refer to these without needing to be aware of how
       Buck is storing data on the disk mid-build.</dd>
+
+      <dt><code>$(maven_coords /&#x2F;path/to:target)</code></dt>
+      <dd>Expands to the Maven coordinates for the build rule. This allows
+      you to access the Maven coordinates for Maven-aware build rules. The
+      format of the expansion is: <code>group_id:artifact_id:version</code>
```
Comment (sdwilsh) [inline docs/__genrule_common.soy]: In the build files, we use a buildr format, so you'd have the artifact type before the version.  Why did you decide to drop that here?

Comment (bocon13) [inline docs/__genrule_common.soy]: @sdwilsh The artifact type (extension) and classifier are not explicitly omitted from the expander. I can extend the documentation to contain the full format: 
<groupId>:<artifactId>[:<extension>[:<classifier>]]:<version>

File: docs/__genrule_common.soy [hunk 2]
```diff
@@ -60,6 +60,14 @@
       <dd>Expands to the location of the output of the build rule. This
       means that you can refer to these without needing to be aware of how
       Buck is storing data on the disk mid-build.</dd>
+
+      <dt><code>$(maven_coords /&#x2F;path/to:target)</code></dt>
+      <dd>Expands to the Maven coordinates for the build rule. This allows
+      you to access the Maven coordinates for Maven-aware build rules. The
+      format of the expansion is: <code>
+      &lt;groupId&gt;:&lt;artifactId&gt;[:&lt;extension&gt;[:&lt;classifier&gt;]]:&lt;version&gt;
```
Comment (sdwilsh) [inline docs/__genrule_common.soy]: Note: you can use the `{literal}` tag here to use actual `<` and `>` so this is more readable

**Comment by sdwilsh** [comment]
Apologies for the delay.  This fell off my radar.

**Comment by alisdair04** [comment]
@facebook-github-bot shipit

File: docs/__genrule_common.soy (hunk 2 above)
Comment (bocon13) [inline docs/__genrule_common.soy]: It seems like the `{literal}` tag blocks HTML escaping, and that `<` and `>` are not auto-escaped within the HTML template without `{literal}`. However, we can make things more readable with a `{let}` block, which should render/escape the angle brackets on replacement. I will push an update.

**Comment by bocon13** [comment]
@sdwilsh No worries! Thanks

**Comment by alisdair04** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**